core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/probabilistic/kde/KDE400EndpointModel.kt

```diff
@@ -174,18 +172,19 @@ class KDE400EndpointModel (
                 M2[j] += delta * delta2
             }
 
-            // Store sample (unbounded or reservoir downsample)
-            if (maxStored <= 0) {
+            /**
+             * Reservoir-style replacement to avoid memory issue
+             *  - Fill up until we reach maxStoredSamples.
+             *  - After that, replace existing samples with decreasing probability
+             *    to maintain a uniform random subset of all seen data.
+             */
+            if (samples.size < maxStoredSamples) {
                 samples.add(x)
             } else {
-                if (samples.size < maxStored) {
-                    samples.add(x)
-                } else {
-                    // reservoir: replace with decreasing probability
-                    val r = kotlin.random.Random.nextLong(seen)
-                    if (r < maxStored) {
-                        samples[r.toInt()] = x
-                    }
+                // reservoir: replace with decreasing probability
+                val r = kotlin.random.Random.nextLong(seen)
```

Comment:
cannot access `kotlin.random.Random` directly. 
all sources of randomness must come from the same singleton `Randomness`